pdk/1.0/modules/ROOT/pages/policies-pdk-configure-features-jwt.adoc

```diff
@@ -0,0 +1,660 @@
+= Using JWT Library Functions
+ifndef::env-site,env-github[]
+include::_attributes.adoc[]
+endif::[]
+:imagesdir: ../assets/images
+
+NOTE: To view an example policy project that uses PDK's JWT Library, see https://github.com/mulesoft/pdk-custom-policy-examples/blob/main/jwt-validation/README.md[JWT Validation Policy Example^].
+
+JSON Web Token (JWT) is a URL-secure method of representing claims to be transferred between two parties. The JWT token contains claims encoded in a JSON object as either the payload of a JSON Web Signature (JWS) or as a JSON web encryption (JWE) structure in plain text. This enables the claims to be digitally signed and protected with a message authentication code (MAC). Because the token is signed, you can trust the information and its source.
+
+The Flex Gateway Policy Development Kit (PDK) JWT library provides a set of tools to validate the JWT signatures and to extract and validate the claims of all incoming requests.
+
+== Token extraction
+
+To extract the JWT token, the library provides a `TokenProvider` enum that has two variants:
+
+* `BearerProvider`: The default `TokenProvider` that extracts tokens with the format `Bearer <token>`.
+* `CustomProvider`: The custom `TokenProvider` that enables you to define a DataWeave expressions to extract the token.
+
+[source,Rust]
+----
+pub enum TokenProvider {
+    #[default]
+    BearerProvider,
+    CustomProvider {
+        expression: Script,
+    },
+}
+----
+
+Both providers implement the `TokenProvision` trait:
+
+[source,Rust]
+----
+pub trait TokenProvision {
+    fn provide(&self, handler: &dyn AttributesBinding) -> Result<String, JWTError>;
+}
+----
+
+=== Bearer Provider
+
+The following example demonstrates how to extract tokens using the `BearerProvider`:
+
+[source,Rust]
+----
+use pdk::hl::*;
+use pdk::jwt::*;
+use pdk::script::HandlerAttributesBinding;
+// [...]
+
+async fn filter(
+    state: RequestState,
+) -> Flow<()> {
+    let headers_state = state.into_headers_state().await;
+
+    // Extract token
+    let token = TokenProvider::BearerProvider
+        .provide(&HandlerAttributesBinding::partial(headers_state.handler()));
+// [...]
+----
+
+=== Custom Provider
+
+To extract a custom token:
+
+. Define a DataWeave expression in the configuration parameters. For example, the following YAML snippet defines the `tokenExtractor` property:
++
+[source,yaml]
+----
+---
```

Comment:
```suggestion

```
I think this line might cause rendering issues, please check in a local build and remove it if the (---) is unnecessary.